aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -142,13 +151,45 @@ unify_from_right(optional<DimnameList> names, optional<DimnameList> other_names)
 
 namespace namedinference {
 
-optional<std::vector<Dimname>> erase_name(optional<DimnameList> self_names, int64_t dim) {
-  if (self_names == nullopt) {
-    return nullopt;
+static std::bitset<64> compute_included_idxs(IntArrayRef excluded_idxs) {
+  std::bitset<64> included_idxs;
+  for (auto i : excluded_idxs) {
+    TORCH_INTERNAL_ASSERT(
+        i < 64,
```

Comment:
hmm... I can imagine it's not common to have a tensor with so many dims, but seems like an odd restriction to impose. Is this a documented somewhere? if we don't have anything yet, we might want to start adding named tensor to the docs.

will this fail fast when constructing a tensor with too many dimensions, or only when performing certain operations that propagate names? If we don't support >64 dims, we should probably fail as soon as it's constructed.

Reply:
TensorIterator has this limitation for ReductionOps: https://github.com/pytorch/pytorch/blob/97a604ef57c15bfcba318d9db4af3b7b17b76a6d/aten/src/ATen/native/ReduceOps.cpp#L40-L50. It is similarly undocumented:
```
In [1]: import torch

In [2]: sizes = [1]*65

In [3]: x = torch.randn(sizes)

In [6]: x.sum(64)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-6375767cd187> in <module>()
----> 1 x.sum(64)

RuntimeError: bitset::set: __position (which is 64) >= _Nb (which is 64)
```

and I am following its behavior.

Since named tensors is still in an experimental state, I want to focus on the implementation coverage before writing up docs. I put up an issue for the TensorIterator limitation and I agree we should have docs for that as well: #23159

Reply:
Ah, interesting. good to know. Maybe we should add a global constant for `TENSOR_MAX_DIMS`. Then in places where it's used it's easier to find the definition and reference/comments/docs.

Reply:
We could use the constant in WrapDimMultiUtils.h.